crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -135,10 +136,14 @@ impl MultiFarming {
 
         // Start syncing
         if dsn_sync {
+            let sync_range_size = PieceIndexHashNumber::MAX / total_pieces * 1024; // 4M per stream
```

Comment:
There is no guarantee this will not result in 20M of data though. I 'd add a TODO that this is not a long-term solution. Instead of ranges we need to operate with number of pieces.